pygmt/gridops.py

```diff
@@ -113,3 +113,119 @@ def grdcut(grid, **kwargs):
+           I="increment",
+           N="nans",
+           R="region",
+           T="toggle",
+           V="verbose",
+           f="colinfo"
+           )
+@kwargs_to_strings(R="sequence")
+
+def grdfilter(grid, **kwargs):
+    """
+    filter a grid file in the time domain using one of the selected convolution
+    or non-convolution isotropic or rectangular filters and compute distances
+    using Cartesian or Spherical geometries. The output grid file can optionally
+    be generated as a sub-region of the input (via *region*) and/or with new increment
+    (via *spacing*) or registration (via *toggle*). In this way, one may have “extra space” in
+    the input data so that the edges will not be used and the output can be within
+    one half-width of the input edges. If the filter is low-pass, then the output
+    may be less frequently sampled than the input.
+        
+    Parameters
+    ----------
+    grid : str or xarray.DataArray
+        The file name of the input grid or the grid loaded as a DataArray.
+        outgrid : str or None
+        The name of the output netCDF file with extension .nc to store the grid
+        in.
+    {F} : str
+        Name of filter type you which to apply, followed by the width
+        b: Box Car; c: Cosine Arch; g: Gaussian; o: Operator; m: Median; p: Maximum Likelihood probability; h: histogram
+        Example: 'm600' for a median filter with width of 600
+    {D}: str
+        Distance flag, that tells how grid (x,y) rrlated to the filter width as follows:
+        flag = p: grid (px,py) with width an odd number of pixels; Cartesian distances.
+        
+        flag = 0: grid (x,y) same units as width, Cartesian distances.
+        
+        flag = 1: grid (x,y) in degrees, width in kilometers, Cartesian distances.
+        
+        flag = 2: grid (x,y) in degrees, width in km, dx scaled by cos(middle y), Cartesian distances.
+        
+        The above options are fastest because they allow weight matrix to be computed only once. The next three options are slower because they recompute weights for each latitude.
+        
+        flag = 3: grid (x,y) in degrees, width in km, dx scaled by cosine(y), Cartesian distance calculation.
+        
+        flag = 4: grid (x,y) in degrees, width in km, Spherical distance calculation.
+        
+        flag = 5: grid (x,y) in Mercator -Jm1 img units, width in km, Spherical distance calculation.
+    
+    {I}: str
+        x_inc [and optionally y_inc] is the grid spacing.
+        (http://docs.generic-mapping-tools.org/latest/grdfilter.html#i)
+    {N}: Str or Number
+        Determine how NaN-values in the input grid affects the filtered output.
+        Values are i|p|r (http://docs.generic-mapping-tools.org/latest/grdfilter.html#n)
+    {R}: Str or list or GMTgrid
+        Specify the region of interest. Set to data minimum BoundinBox if not provided.
+        (http://docs.generic-mapping-tools.org/latest/gmt.html#r-full)
```

Comment:
For common options (**-R** for GMT CLI and *region* for PyGMT), you can use `{R}`, it will be replaced by the docstring for *region*. You can check how the docstring is written in https://github.com/GenericMappingTools/pygmt/blob/master/pygmt/gridops.py#L31
```suggestion
    {R}
```